R/build_leanpub_embed_files.R

```diff
@@ -0,0 +1,41 @@
+# script to fix embedded files for Leanpub
+# originally used the raw rmd files before conversion  with bookdown_to_leanpub() - now part of the function at the beginning
+
+library(here)
+library(stringr)
+
+
+split_embed_knitr<-function(to_replace_embed) {
+  # grab the second set of text in quotes
+  link_url <-strsplit(to_replace_embed, "\\\"")[[1]][2]
+  y =  paste0("Check out this [link](", link_url,").")
+  y}
+
+split_embed_html<-function(to_replace_embed) {
+  # grab contents after src=
+  link_url <-strsplit(to_replace_embed, "src=")[[1]][2]
+  # only keep the part after "\" and bfore the next element
+  link_url <-strsplit(link_url, "\\\"")[[1]][2]
+  y =  paste0("Check out this [link](", link_url,").")
+  y}
+
+# to find rmd files to modify
+files<-list.files(here(), pattern = ".rmd|.Rmd|.RMd|.RMD")
+
+#code to modify the lines within the rmd about the embedded files - to instead create a link
+for(i in files){
```

Comment:
I think ideally this would be a function too. I can work on functional-izing this.

Reply:
this could be used for a function to add text about feedback...